crates/router/tests/connectors/cryptopay.rs

```diff
@@ -0,0 +1,149 @@
+                    last_name: Some(Secret::new("last".to_string())),
+                    line1: Some(Secret::new("line1".to_string())),
+                    line2: Some(Secret::new("line2".to_string())),
+                    city: Some("city".to_string()),
+                    zip: Some(Secret::new("zip".to_string())),
+                    country: Some(api_models::enums::CountryAlpha2::IN),
+                    ..Default::default()
+                }),
+                phone: Some(api::PhoneDetails {
+                    number: Some(Secret::new("1234567890".to_string())),
+                    country_code: Some("+91".to_string()),
+                }),
+            }),
+            ..Default::default()
+        }),
+        return_url: Some(String::from("https://google.com")),
+        ..Default::default()
+    })
+}
+
+fn payment_method_details() -> Option<types::PaymentsAuthorizeData> {
+    Some(types::PaymentsAuthorizeData {
+        amount: 1,
+        currency: enums::Currency::USD,
+        payment_method_data: types::api::PaymentMethodData::Crypto(CryptoData {
+            pay_currency: Some("XRP".to_string()),
+        }),
+        confirm: true,
+        statement_descriptor_suffix: None,
+        statement_descriptor: None,
+        setup_future_usage: None,
+        mandate_id: None,
+        off_session: None,
+        setup_mandate_details: None,
+        browser_info: None,
+        order_details: None,
+        order_category: None,
+        email: None,
+        payment_experience: None,
+        payment_method_type: None,
+        session_token: None,
+        enrolled_for_3ds: false,
+        related_transaction_id: None,
+        router_return_url: Some(String::from("https://google.com/")),
+        webhook_url: None,
+        complete_authorize_url: None,
+        capture_method: None,
+        customer_id: None,
+    })
+}
+
+// Creates a payment using the manual capture flow
+#[actix_web::test]
+async fn should_only_authorize_payment() {
+    let response = CONNECTOR
+        .authorize_payment(payment_method_details(), get_default_payment_info())
+        .await
+        .expect("Authorize payment response");
+    assert_eq!(response.status, enums::AttemptStatus::AuthenticationPending);
+    let resp = response.response.ok().unwrap();
+    let endpoint = match resp {
+        types::PaymentsResponseData::TransactionResponse {
+            redirection_data, ..
+        } => Some(redirection_data),
+        _ => None,
+    };
+    assert!(endpoint.is_some())
+}
+
+// Synchronizes a successful transaction.
+#[actix_web::test]
+async fn should_sync_authorized_payment() {
+    let response = CONNECTOR
+        .psync_retry_till_status_matches(
+            enums::AttemptStatus::Authorized,
+            Some(types::PaymentsSyncData {
+                connector_transaction_id: router::types::ResponseId::ConnectorTransactionId(
+                    "ea684036-2b54-44fa-bffe-8256650dce7c".to_string(),
```

Comment:
Are these hardcoded transaction IDs in this function and `should_sync_unresolved_payment()` intentional?

Reply:
Hey, yes they are intentional. These are the connector transaction id stuck in specific status.